src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -301,8 +301,16 @@ private Object parseValue(StaplerRequest request) throws ServletException, IOExc
             }
         }
 
-        // TODO: perhaps we should return a different object to allow the workflow to look up
-        // who approved it, etc?
+        // If requested, add the approver to the list of returned parameters.
+        String submitter = input.getSubmitter();
+        if (submitter != null) {
```

Comment:
This means that if you write

```groovy
input message: 'x', submitterParameter: 'user'
```

you will silently get a null return value. That seems wrong to me—you may want to know who the submitter was, even if you are not restricting them in any particular way.

I think this `if`-block is just left over from the previous mistake, and you can simply ignore `submitter` here.

(Even if it *is* intentional to only allow `submitterParameter` when `submitter` is also set, this should be emphasized in HTML documentation, and checked in form validation in `InputStep.DescriptorImpl`.)